### PR TITLE
Sign up feature

### PR DESCRIPTION
**Pull request by arhussain1**
-Added a script to the package.json file to allow for single cypress file testing
-Added required statements to email and password inputs on the sign up page. This ensures users cannot create accounts with missing information
-Added userExists variables in user controllers create method which checks if a users email attempting to sign up already exists in the mongo database i.e. A user cannot sign up if the email is already in use.

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -5,14 +5,23 @@ const UsersController = {
...
+    if (userExists == false) {
+      user.save((err) => {
+        if (err) {
+          throw err;
+        }
+        res.status(201).redirect("/posts");
+      });
+    } else {
+      // should flash an error message back to the user
+      res.redirect("/users/new");
```
Comment (HarryClenyg) [inline controllers/users.js]: we think this should redirect the existing user to sessions/new (the log in page).  This looks like it's sending the existing user to the new user page.

File: controllers/users.js [hunk 2]
```diff
@@ -5,14 +5,23 @@ const UsersController = {
...
-    });
+
+    const userExists = await User.exists({email: user.email})
+
+    if (userExists == false) {
+      user.save((err) => {
+        if (err) {
+          throw err;
+        }
+        res.status(201).redirect("/posts");
```
Comment (HarryClenyg) [inline controllers/users.js]: there is a test for 'user signs up and is redirected to sign in'.  Here it looks like when the user signs up they are redirected to posts?

File: cypress/integration/user_can_sign_up_spec.js [hunk 3]
```diff
@@ -1,11 +1,59 @@
...
   });
+
+  // user should get an error if either or both field is empty 
+  it("A user should get an error if the email field is empty", () => {
+    cy.visit("/users/new");
+    cy.get("#password").type("password");
+    cy.get("#createAccount").click();
+
+    cy.url().should("include", "/users/new");
+  })
```
Comment (HarryClenyg) [inline cypress/integration/user_can_sign_up_spec.js:23]: shall we make a new ticket for testing for this error?

File: cypress/integration/user_can_sign_up_spec.js [hunk 4]
```diff
@@ -1,11 +1,59 @@
...
+    // second sign up
+    cy.visit("/users/new");
+    cy.get("#email").type("someone@example.com");
+    cy.get("#password").type("password");
+    cy.get("#createAccount").click();
+
+    cy.url().should("include", "/users/new");
+    // expect error to pop up on screen
+    //cy.get(".err").should("contain", "This email is in use");
+
```
Comment (HarryClenyg) [inline cypress/integration/user_can_sign_up_spec.js:57]: make a new ticket for all error flashes

File: controllers/users.js (hunk 2 above)
Comment (arhussain1) [reply controllers/users.js]: Hey Harry that branch of the if statement is for when the user fails to sign up i.e. the users emails is already registered. 

So we would want him to be sent back to the sign up page and ideally in the future flash him a message telling him to use another email address

However, should line 18 redirect to sessions/new

File: controllers/users.js (hunk 1 above)
Comment (arhussain1) [reply controllers/users.js]: Same reason as above

**Review by HarryClenyg — APPROVED** [review]

**Pull request merged**